mantidproject/mantid
# Create script to produce DragNDrop package based on Conda build outputs

**Description of work.**

Shipping a full standalone macOS package is still very important for many of our users.
This work creates a new script to create the dmg bundle from existing Conda packages
rather than packaging up our standard CMake build as is currently done.
This has the advantage that we do not need separate CMake code to support the full package
as this will just inherit any fixes that are applied to the Conda work.

Some future work would be attempt to slim down the bundle even further but this work creates
a working bundle and forms a good basis to move forward.



**To test:**

This work does not require a build of mantid just a checkout of this branch to access the new script.
It can be tested using the standard build directory though.
To test:

* Switch to the build directory
* Run `<source_dir>/installers/conda/osx/create_bundle.sh -c mantid/label/nightly MantidWorkbenchNightly <source_dir>/images/mantid_workbenchnightly.icns`. `<source_dir>` is the root of your git checkout.
* The .dmg will appear in your current directory.
* Delete the `_bundle_build` directory that was created
* Open the dmg, install it and run it.

Fixes #33024 

*This does not require release notes* because **nothing is yet being release to users here. This adds the mechanism to build the new packages but not the whole pipeline.**



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
